src/main/java/games/strategy/util/IntegerMap.java

```diff
@@ -14,7 +14,7 @@
  *
  * @param <T> The type of the map key.
  */
-public class IntegerMap<T> implements Cloneable, Serializable {
+public final class IntegerMap<T> implements Cloneable, Serializable {
```

Comment:
The addition of the `final` modifier is to ensure the contract for `equals()` and `hashCode()` cannot be violated if another class were to extend this class.  As there are currently no subclasses of `IntegerMap`, I just made the entire class `final`.

This does not affect serialization.  Only the class name and `serialVersionUID` are serialized via `ObjectStreamClass`.  Modifiers such as `final` are not recorded in the serialization stream.

(This comment applies equally to `LinkedIntegerMap`.)